cirq-ft/cirq_ft/algos/qrom_test.py

```diff
@@ -116,6 +116,76 @@ def test_t_complexity(data):
     assert cirq_ft.t_complexity(g.gate).t == max(0, 4 * n - 8), n
 
 
+def _assert_qrom_has_diagram(qrom: cirq_ft.QROM, expected_diagram: str):
+    gh = cirq_ft.testing.GateHelper(qrom)
+    op = gh.operation
+    context = cirq.DecompositionContext(qubit_manager=cirq_ft.GreedyQubitManager(prefix="anc"))
+    circuit = cirq.Circuit(cirq.decompose_once(op, context=context))
+    selection = [
+        *itertools.chain.from_iterable(gh.quregs[reg.name] for reg in qrom.selection_registers)
+    ]
+    selection = [q for q in selection if q in circuit.all_qubits()]
+    anc = sorted(set(circuit.all_qubits()) - set(op.qubits))
+    selection_and_anc = (selection[0],) + sum(zip(selection[1:], anc), ())
+    qubit_order = cirq.QubitOrder.explicit(selection_and_anc, fallback=cirq.QubitOrder.DEFAULT)
+    cirq.testing.assert_has_diagram(circuit, expected_diagram, qubit_order=qubit_order)
+
+
+def test_qrom_variable_spacing():
+    # Tests for variable spacing optimization applied from https://arxiv.org/abs/2007.07391
+    data = [1, 2, 3, 4, 5, 5, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8]  # Figure 3a.
+    assert cirq_ft.t_complexity(cirq_ft.QROM.build(data)).t == 6 * 4
```

Comment:
Is it possible to add a test for the negative case they give in the paper. As in, " For example, we can use
a region such as {4, 5}, but not {3, 4}, because 4 ≡ 100 and 5 ≡ 101, so grouping 4 and 5 corresponds to ignoring the
least significant bit, but the least significant bit changes between 3 and 4."

Reply:
added the test